packages/esm-patient-chart-app/src/visit/visit-prompt/end-visit.component.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import styles from './end-visit.component.scss';
+import ComposedModal, { ModalHeader, ModalBody } from 'carbon-components-react/es/components/ComposedModal';
+import Button from 'carbon-components-react/es/components/Button';
+import { useTranslation } from 'react-i18next';
+import { getStartedVisit, showToast, updateVisit, useVisit } from '@openmrs/esm-framework';
+import dayjs from 'dayjs';
+
+interface EndVisitPromptProps {
+  patientUuid: string;
+  openModal: boolean;
+  closeModal: () => void;
+}
+
+const EndVisitPrompt: React.FC<EndVisitPromptProps> = ({ patientUuid, openModal, closeModal }) => {
+  const { t } = useTranslation();
+  const dateFormat = 'DD - MMM - YYYY';
+
+  const { currentVisit } = useVisit(patientUuid);
+
+  const endCurrentVisit = () => {
+    const endVisitPayload = {
+      location: currentVisit.location.uuid,
+      startDatetime: currentVisit.startDatetime,
+      visitType: currentVisit.visitType.uuid,
+      stopDatetime: new Date(),
+    };
+    const ac = new AbortController();
+    const sub = updateVisit(currentVisit.uuid, endVisitPayload, ac).subscribe(
+      (response) => {
+        if (response.status === 200) {
+          getStartedVisit.next(null);
+          closeModal();
+          showToast({
+            kind: 'success',
+            description: t('visitEndSuccessfully', 'Ended current visit successfully'),
+          });
+        }
+      },
+      (error) => {
+        showToast({
+          kind: 'error',
+          description: t('visitEndError', 'Error ending current visit'),
+        });
+      },
+    );
+
+    return () => sub && sub.unsubscribe();
```

Comment:
Similar problem to the memory leak from before, since `endCurrentVisit()()` (i.e. the returned function) is never going to be called. You can probably solve these in the same way, however you figure out to do it.